frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt

```diff
@@ -1,123 +1,309 @@
 package com.virtuslab.gitmachete.frontend.actions.dialogs
 
+import com.intellij.dvcs.DvcsUtil
+import com.intellij.ide.ui.laf.darcula.DarculaUIUtil
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.ui.DialogWrapper
 import com.intellij.openapi.ui.ValidationInfo
-import com.intellij.ui.DocumentAdapter
-import com.intellij.ui.layout.CellBuilder
-import com.intellij.ui.layout.ValidationInfoBuilder
-import com.intellij.ui.layout.panel
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.MutableCollectionComboBoxModel
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread
+import com.intellij.util.ui.JBUI
 import com.virtuslab.branchlayout.api.IBranchLayout
 import com.virtuslab.branchlayout.api.IBranchLayoutEntry
 import com.virtuslab.gitmachete.frontend.actions.common.SlideInOptions
 import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString
-import java.awt.event.KeyEvent
+import git4idea.branch.GitBranchUtil
+import git4idea.commands.Git
+import git4idea.commands.GitCommand
+import git4idea.commands.GitLineHandler
+import git4idea.merge.GitMergeDialog
+import git4idea.repo.GitRepository
+import git4idea.repo.GitRepositoryManager
+import git4idea.ui.ComboBoxWithAutoCompletion
+import java.awt.Insets
+import java.util.regex.Pattern
 import javax.swing.JCheckBox
+import javax.swing.JLabel
+import javax.swing.JPanel
 import kotlin.apply
-import kotlin.text.isEmpty
-import kotlin.text.trim
+import net.miginfocom.layout.AC
```

Comment:
miglayout is baked into IntelliJ? 😯

Reply:
(as I suppose it's this stuff https://www.miglayout.com/)

Reply:
yup